ablySpec/RealtimeClientConnection.swift

```diff
@@ -1302,6 +1302,63 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN15
+            context("connection failures once CONNECTED") {
+
+                // RTN15a
+                pending("should be able to catch up by receiving the queued ProtocolMessages if the connection is disconnected unexpectedly") {
+                    let options = AblyTests.commonAppSetup()
+                    options.disconnectedRetryTimeout = 1.0
+
+                    let client1 = ARTRealtime(options: options)
+                    defer { client1.close() }
+                    let channel1 = client1.channels.get("test")
+
+                    let client2 = ARTRealtime(options: options)
+                    defer { client2.close() }
+                    let channel2 = client2.channels.get("test")
+
+                    var reconnected = false
+                    channel1.subscribe { message in
+                        if !reconnected {
+                            fail("Shouldn't receive the messsage")
+                        }
+                    }
+
+                    expect(channel1.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+
+                    client1.onDisconnected()
+
+                    channel2.publish(nil, data: "message") { errorInfo in
+                        expect(errorInfo).to(beNil())
+                    }
+
+                    waitUntil(timeout: options.disconnectedRetryTimeout) { done in
+                        client1.connection.on { stateChange in
+                            switch stateChange!.current {
+                            case .Connecting:
+                                expect(client1.resuming).to(beTrue())
+                                done()
+                            case .Connected:
+                                expect(client1.resuming).to(beFalse())
+                                reconnected = true
+                            default:
+                                break
+                            }
+                        }
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        // Check the queued message
+                        channel1.subscribe { message in
```

Comment:
Why are you resubscribing, you have a listener already defined above on line 1322. Also, subscribing after you're connected is not going to work because it's quite possible the message has already been received at this point.